astacus/manifest.py

```diff
@@ -0,0 +1,111 @@
+    p_manifest.add_argument("-s", "--storage", type=str, help="Storage to use")
+    manifest_subparsers = p_manifest.add_subparsers(title="Manifest commands")
+    create_list_parser(manifest_subparsers)
+    create_describe_parser(manifest_subparsers)
+    create_dump_parser(manifest_subparsers)
+
+
+def create_list_parser(subparsers):
+    p_list = subparsers.add_parser("list", help="List backup manifests in object storage")
+    p_list.set_defaults(func=_run_list)
+
+
+def create_describe_parser(subparsers):
+    p_describe = subparsers.add_parser("describe", help="Print info/statistics from a backup manifest")
+    p_describe.add_argument("manifest", type=str, help="Manifest object name (can be obtained by running manifest list)")
+    p_describe.set_defaults(func=_run_describe)
+
+
+def create_dump_parser(subparsers):
+    p_dump = subparsers.add_parser("dump", help="Dump contents of the manifest to standard output")
+    p_dump.add_argument("-i", "--indent", type=int, help="How many spaces to use per tab in json output")
+    p_dump.add_argument("-F", "--files", action="store_true", default=False, help="Dump files in snapshot results")
+    p_dump.add_argument("-H", "--hashes", action="store_true", default=False, help="Dump hashes in snapshot results")
+    p_dump.add_argument("manifest", type=str, help="Manifest object name (can be obtained by running manifest list)")
+    p_dump.set_defaults(func=_run_dump)
+
+
+def _run_list(args):
+    rohmu_storage = _create_rohmu_storage(args.config, args.storage)
+    json_names = rohmu_storage.list_jsons()
+    for _json in json_names:
+        print(_json)
+
+
+def _run_describe(args):
+    rohmu_storage = _create_rohmu_storage(args.config, args.storage)
+    manifest = rohmu_storage.download_json(args.manifest, ipc.BackupManifest)
+    print(manifest.plugin, "manifest", args.manifest)
+    print("===============================")
+    print("Started at:", manifest.start.isoformat())
+    print("Finished at:", manifest.end.isoformat())
+    print("Attempts:", manifest.attempt)
+    for i, snapshot_result in enumerate(manifest.snapshot_results):
+        print(
+            f"Snapshot #{i} created at {snapshot_result.start.isoformat()}",
+            f"contains {snapshot_result.files} files ({snapshot_result.total_size} bytes)",
+        )
+    for i, upload_result in enumerate(manifest.upload_results):
+        print(f"Node #{i} uploaded {upload_result.total_stored_size} bytes out of {upload_result.total_size}")
+    if manifest.tiered_storage_results:
+        print(
+            f"References {manifest.tiered_storage_results.n_objects} objects in tiered storage",
+            f"totalling {manifest.tiered_storage_results.total_size_bytes} bytes",
+        )
+
+
+def _run_dump(args):
+    rohmu_storage = _create_rohmu_storage(args.config, args.storage)
+    with rohmu_storage.open_json_bytes(args.manifest) as mapped_file:
```

Comment:
How is performance in general here?

if indent is 0 and we are not redacting, we can just copy the contents of the file to stdout.  Maybe for very large manifests this would be a good performance optimization.

Reply:
good point, changed to "dump to stdout without parsing" by default, set all the formatting to be an optional extra